src/bsp/platform/freertos-posix/iotc_bsp_io_net_posix.c

```
@@ -33,57 +35,71 @@ extern "C" {
-
-  if (-1 == *iotc_socket) {
+iotc_bsp_io_net_state_t
+iotc_bsp_io_net_socket_connect(iotc_bsp_socket_t* iotc_socket, const char* host,
+                               uint16_t port,
+                               iotc_bsp_socket_type_t socket_type) {
+  struct addrinfo hints;
+  struct addrinfo *result, *rp = NULL;
+  int status;
+
+  memset(&hints, 0, sizeof(hints));
+  hints.ai_family = AF_UNSPEC;
+  hints.ai_socktype = socket_type;
+  hints.ai_flags = 0;
+  hints.ai_protocol = 0;
+
+  // Address resolution
+  status = getaddrinfo(host, NULL, &hints, &result);
+  if (0 != status) {
     return IOTC_BSP_IO_NET_STATE_ERROR;
   }
 
-  /* Enable nonblocking mode for a posix socket */
-  const int flags = fcntl(*iotc_socket, F_GETFL, 0);
+  for (rp = result; rp != NULL; rp = rp->ai_next) {
+    *iotc_socket = socket(rp->ai_family, rp->ai_socktype, rp->ai_protocol);
+    if (-1 == *iotc_socket)
+      continue;
 
-  if (-1 == flags || -1 == fcntl(*iotc_socket, F_SETFL, flags | O_NONBLOCK)) {
-    return IOTC_BSP_IO_NET_STATE_ERROR;
-  }
-
-  return IOTC_BSP_IO_NET_STATE_OK;
-}
-
-iotc_bsp_io_net_state_t iotc_bsp_io_net_connect(iotc_bsp_socket_t* iotc_socket,
-                                                const char* host,
-                                                uint16_t port) {
-  struct hostent* hostinfo = gethostbyname(host);
+    // Set the socket to be non-blocking
+    const int flags = fcntl(*iotc_socket, F_GETFL);
+    if (-1 == fcntl(*iotc_socket, F_SETFL, flags | O_NONBLOCK)) {
+      freeaddrinfo(result);
+      return IOTC_BSP_IO_NET_STATE_ERROR;
+    }
 
-  /* if null it means that the address has not been found */
-  if (NULL == hostinfo) {
-    return IOTC_BSP_IO_NET_STATE_ERROR;
-  }
+    switch (rp->ai_family) {
+    case AF_INET6:
+      ((struct sockaddr_in6*)(rp->ai_addr))->sin6_port = htons(port);
+      break;
+    case AF_INET:
+      ((struct sockaddr_in*)(rp->ai_addr))->sin_port = htons(port);
+      break;
```

default clause should return an error.